### PR TITLE
Removed generate_bins_generic

### PR DESCRIPTION
**Pull request by WillAyd**
appears to be dead groupby code

### CODE REVIEW

File: pandas/tests/groupby/test_bin_groupby.py [hunk 1]
```diff
@@ -43,42 +41,21 @@ def test_series_bin_grouper():
...
-        self.bins = np.array([3, 6], dtype=np.int64)
+def test_generate_bins():
+    values = np.array([1, 2, 3, 4, 5, 6], dtype=np.int64)
+    binner = np.array([0, 3, 6, 9], dtype=np.int64)
 
-    def test_generate_bins(self):
-        values = np.array([1, 2, 3, 4, 5, 6], dtype=np.int64)
-        binner = np.array([0, 3, 6, 9], dtype=np.int64)
+    bins = lib.generate_bins_dt64(values, binner, closed="left")
+    assert (bins == np.array([2, 5, 6])).all()
```
Comment (simonjayhawkins) [inline pandas/tests/groupby/test_bin_groupby.py]: might be worth changing to tm.assert_numpy_array_equal while changing this

File: pandas/tests/groupby/test_bin_groupby.py [hunk 2]
```diff
@@ -43,42 +41,21 @@ def test_series_bin_grouper():
     assert_almost_equal(counts, exp_counts)
 
 
-class TestBinGroupers:
-    def setup_method(self, method):
-        self.obj = np.random.randn(10, 1)
-        self.labels = np.array([0, 0, 0, 1, 1, 1, 2, 2, 2, 2], dtype=np.int64)
-        self.bins = np.array([3, 6], dtype=np.int64)
+def test_generate_bins():
```
Comment (simonjayhawkins) [inline pandas/tests/groupby/test_bin_groupby.py]: can this be parametrized left/right and binner ?

File: pandas/tests/groupby/test_bin_groupby.py [hunk 3]
```diff
@@ -43,42 +41,21 @@ def test_series_bin_grouper():
...
 
-        for func in [lib.generate_bins_dt64, generate_bins_generic]:
-            bins = func(values, binner, closed="left")
-            assert (bins == np.array([2, 5, 6])).all()
+    bins = lib.generate_bins_dt64(values, binner, closed="right")
+    assert (bins == np.array([3, 6, 6])).all()
 
-            bins = func(values, binner, closed="right")
-            assert (bins == np.array([3, 6, 6])).all()
+    values = np.array([1, 2, 3, 4, 5, 6], dtype=np.int64)
```
Comment (simonjayhawkins) [inline pandas/tests/groupby/test_bin_groupby.py]: is this a repeat of L45

File: pandas/tests/groupby/test_bin_groupby.py (hunk 2 above)
Comment (simonjayhawkins) [inline pandas/tests/groupby/test_bin_groupby.py]:
```suggestion
def test_generate_bins_dt64():
```

**Review by simonjayhawkins — COMMENTED** [review]
@WillAyd generally looks good. minor comments here or follow-on

**Comment by jbrockmendel** [comment]
nice!

File: pandas/tests/groupby/test_bin_groupby.py [hunk 4]
```diff
@@ -6,9 +6,8 @@
 from pandas.core.dtypes.common import ensure_int64
 
 from pandas import Index, Series, isna
-from pandas.core.groupby.ops import generate_bins_generic
 import pandas.util.testing as tm
-from pandas.util.testing import assert_almost_equal
+from pandas.util.testing import assert_almost_equal, assert_numpy_array_equal
```
Comment (simonjayhawkins) [inline pandas/tests/groupby/test_bin_groupby.py]: see #23018

**Review by simonjayhawkins — APPROVED** [review]
@WillAyd lgtm

**Comment by jbrockmendel** [comment]
@WillAyd can you rebase; hopefully azure will go through this time

**Comment by WillAyd** [comment]
Just restarted build instead to see. These intermittent failures have been rather frustrating lately :-X

**Comment by jbrockmendel** [comment]
one more push to restart CI?

**Comment by WillAyd** [comment]
Looks like that one worked

**Comment by WillAyd** [comment]
Thanks @jbrockmendel and @simonjayhawkins for review

**Pull request merged**